src/Collections-Weak-Tests/WeakIdentitySetTest.class.st

```diff
@@ -50,10 +50,25 @@ WeakIdentitySetTest >> testAddIdenticalElements [
 
 { #category : #tests }
 WeakIdentitySetTest >> testAddIncludesSizeReclaim [
-	"It is complex test in superclass to check how elements are garbage collected.
-	And how #includes: behavios after that.
-	The logic is based on WeakSet and it is not compatible to WeakIdentitySet.
-	Adopting this test for WeakIdentitySest is not simple task"
+	| ws o2 o3 |
+	o2 := 1 / 2.
+	o3 := '123' copy.
+	ws := self classToBeTested new.
+	ws add: o2.
+	ws add: o3.
+	self assert: ws size equals: 2.
+	self assert: (ws includes: o2).
+	self assert: (ws includes: o3).
+
+	"only one copy is added"
+	ws add: o3 copy.
+	self assert: ws size equals: 3.
+
+	"reclame objects so that slots of ws are nilled out"
+	o2 := o3 := nil.
+	2 timesRepeat: [ Smalltalk garbageCollect ].
```

Comment:
Hopefully not :)
The thing is: when a single object has many ephemeron subscriptions, only one of them is triggered per GC.
That is because when one ephemeron triggers, it becomes a normal strong object and prevents triggering the other ephemerons (for that same object).

But I'm not against alternatives!! I did this to make the test pass.
Another solution that may be more robust is to GC until the size of the collection is zero.

```smalltalk
[ ws isEmpty ] whileFalse: [ Smalltalk garbageCollect ].
```
This will naturally either:
 - eventually be green
 - or time out for the red